File: typescript/src/app/04-i-interface-segregation/concept/before/CanonPrinter.ts
import AllInOnePrinter from './AllInOnePrinter';

class CanonPrinter implements AllInOnePrinter {
  print(): string {
    return 'Printing done!';
  }
  scan(): string {
    throw new Error('Method not implemented.');
  }
  copy(): string {
    throw new Error('Method not implemented.');
  }
  fax(): string {
    throw new Error('Method not implemented.');
  }
}

export default CanonPrinter;

File: java/src/main/java/isp/before/CanonPrinter.java
package isp.before;

public class CanonPrinter implements AllInOne {
    @Override
    public String print() {
        return "Print done!";
    }

    @Override
    public String scan() {
        // do nothing...
        return null;
    }

    @Override
    public String copy() {
        // do nothing...
        return null;
    }

    @Override
    public String fax() {
        // do nothing...
        return null;
    }
}
